### PR TITLE
Port to tf2  .replace("tf",  "tf.compat.v1")

### PR DESCRIPTION
**Pull request by benleetownsend**
Things to note.
* Keras optimizers do not increment global step, so we now have to do this.
* Preempting having to change some variable naming, I have added a permit_uninitialized config item. The behaviour of this is that everything under model/featurizer/ needs to be initialized or matched by this regex else an error is raised.

### CODE REVIEW

File: finetune/scheduler.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 import tensorflow as tf
 from finetune.base import BaseModel
+from finetune.custom_ops import BytesInUse, BytesLimit, MaxBytesInUse
```
Comment (madisonmay) [inline finetune/scheduler.py:6]: :100:

File: setup.py [hunk 2]
```diff
@@ -25,6 +27,13 @@
     "tabulate>=0.8.6,<0.9.0", 
 ]
 
+
+class OpsBuild(build_ext):
+    def run(self):
+        script = os.path.join(os.path.dirname(__file__), "finetune", "custom_ops", "build.sh")
+        if subprocess.run(["sh", script]).returncode != 0:
+            warnings.warn("Failed to build the finetune ops, most aspects of finetune should function anyway.")
```
Comment (madisonmay) [inline setup.py]: I would just be super explicit here: 

"Failed to build the finetune memory management ops required for use of `Scheduler`. If you don't intend to use `Scheduler` you can safely ignore this message.  To build the ops later  execute {script}"

File: tests/test_scheduler_memory_fraction.py [hunk 3]
```diff
@@ -50,7 +52,8 @@ def setUpClass(cls):
     @classmethod
     def tearDownClass(self):
         shutil.rmtree("tests/saved-models/")
-        
+
+    @pytest.mark.skip()
```
Comment (madisonmay) [inline tests/test_scheduler_memory_fraction.py:56]: Was this temporary or is there something wrong w/ this test?

Comment (benleetownsend) [reply tests/test_scheduler_memory_fraction.py:56]: This is the test that is kinda awkward because it needs to be run alone. I will see if tf2 has any options we didn't have with tf1, possibly something to do with the logical device config stuff??

File: finetune/encoding/input_encoder.py [hunk 4]
```diff
@@ -8,9 +8,15 @@
 import spacy
 import numpy as np
 
+NLP = None
+
+def get_spacy():
```
Comment (lhz1029) [inline finetune/encoding/input_encoder.py:13]: What's the reasoning behind this change? Were we somehow reloading spacy multiple times?

Comment (lhz1029) [reply finetune/encoding/input_encoder.py:13]: It does actually make mocking this out a lot easier for customv1 though--I think I'll copy this formulation over there :)

Comment (benleetownsend) [reply finetune/encoding/input_encoder.py:13]: Mostly just means we don't load spacy at import or if we don't use it.
